feat: Enable the consumer of the PactVerifier to provide RequestTimeout

mortezaalizadeh opened the pull request:
Also made the constructor public, so values for non-supported pact-reference arguments
can be provided

adamrodger commented on src/PactNet/Verifier/PactVerifier.cs:
```diff
@@ -34,7 +35,7 @@ public PactVerifier(PactVerifierConfig config) : this(new Dictionary<string, str
         /// <param name="config">Pact verifier config</param>
-        internal PactVerifier(IDictionary<string, string> verifierArgs, PactVerifierConfig config)
+        public PactVerifier(IDictionary<string, string> verifierArgs, PactVerifierConfig config)
         {
```
This constructor is deliberately internal only. Users of the library shouldn't be able to construct it.

mortezaalizadeh replied:
@adamrodger  done

adamrodger reviewed (changes requested):
The new method should be added to the appropriate interface and tests should be added. I think this method belongs on the same interface as the one which allows you to set log level

mortezaalizadeh commented:
@adamrodger  I added method to the interface, added test and also fixed the solution file having issue loading Consumer and Consumer.Tests project properly. Please have a look

mortezaalizadeh commented:
@adamrodger  we are currently using local built version of the library (V4.0), so I was wondering if you guys pushing the nuget packages somewhere we can use in our pipeline?

adamrodger commented:
This change will be superseded by the move to the new verifier FFI in #353. Request timeout is part of the verification options call.
